model/private_subnet.rb

```diff
@@ -25,6 +23,13 @@ def hyper_tag_name(project)
 
   include Authorization::TaggableMethods
 
+  def destroy
```

Comment:
No need these custom `destroy` helpers. `association_dependencies` has `many_to_many` support. Yo need to add `firewalls` to the list.

https://sequel.jeremyevans.net/rdoc-plugins/classes/Sequel/Plugins/AssociationDependencies.html

Reply:
```
:many_to_many | :nullify (removes all related entries in join table)
```

Reply:
Yes, but this calls `delete` which we forcefully do not allow...